assets/js/ts_fas_acih.js

```diff
@@ -1237,12 +1231,407 @@ if (pathName === "/home" || pathName === "/home.html" || pathName === "/u" || pa
       threshold: 0.1
    });
 
+   // returns the fully rendered note div or null
+   function renderNote(noteData) {
+      function renderUsername(username, pronouns, time) {
+         const displayDate = timeAgo(time);
+         if (pronouns) {
+            return `@${username} • ${pronouns} • ${displayDate}`;
+         } else {
+            return `@${username} • ${displayDate}`;
+         }
+      }
+
+      // TODO: we actually use fontawesome everywhere so lift this to somewhere toplevel
+      function faIcon(name, sm = false, xs = false) {
+         const icon = document.createElement("i");
+         icon.classList.add("fa-solid");
+         icon.classList.add("fa-" + name);
+         // unsure of the meaning of these but they were there so whatevs
+         if (sm) icon.classList.add("fa-sm");
+         if (xs) icon.classList.add("fa-xs");
+         return icon;
+      }
+
+      if (noteData.id === undefined) return null;
+
+      const noteDiv = document.createElement("div");
+      noteDiv.className = "note";
+      noteDiv.id = noteData.id;
+
+      const hasCw = noteData.isNsfw || noteData.isSensitive || noteData.isPolitical;
+      const cwType = noteData.isNsfw ? "NSFW" : (noteData.isSensitiwe ? "sensitive" : "political");
+
+      if (hasCw) {
+         firebase.auth().onAuthStateChanged(function (user) {
+            if (!user) {
+               // were in a callback so we cant directly return out of the outer function
```

Comment:
Nitpick:
`were in a callback so we cant directly return out of their outer function` -> `we're in a callback so we cant directly return out of their outer function`

Reply:
i mean yeah, right now it's just me and you working on transsocial. but even if i understand it, that doesn’t mean someone else who tries to look at the code later will. just wanna make the codebase clear for anyone who may look at it in the future! :) 
futureproofing, you could say

Reply:
i get your point, but i doubt an apostrophe is gonna make the difference. if someone doesnt understand the code because they read it as the simple past plural of be, i think its on them. 
im willing to add it if you insist (this isnt worth a huge argument) but i really dont think its necessary. 

btw you can suggest a change inline with
\```suggestion
               // we're in a callback so we cant directly return out of the outer function
\```